packages/bot-web-ui/src/pages/tutorials/dbot-tours/onboarding-tour/onboarding-tour-mobile.tsx

```diff
@@ -31,6 +31,23 @@ const OnboardingTourMobile = observer(() => {
     const test_id = tour_step_key === 8 ? 'finish-onboard-tour' : 'next-onboard-tour';
     const hide_prev_button = [1, 2, 8];
     const is_tour_active = active_tour === 'onboarding';
+    
+    const getItemFromLocalStorage = (key: string): Promise<string | null> => {
+        return new Promise(resolve => {
+            const value = getSetting(key);
+            resolve(value);
+        });
+    };
+    const checkTokenForTour = async () => {
+        const token = await getItemFromLocalStorage('onboard_tour_token');
+        if (!token && active_tab === 0) {
+            setActiveTour('onboarding');
+        }
+    };
+
+    React.useEffect(() => {
+        checkTokenForTour();
+    }, [active_tab]);
```

Comment:
@rupato-deriv accessing local storage is a synchronous operation and I think using promise is unnecessary here. Can you please check this part again?